### ISSUE TITLE
Filename and download option being ignored

### ISSUE THREAD

**Issue by T0mi** [FriendsOfCake/CakePdf#254]
Hi guys!

After updating package version from 3.3.1 to 3.4.0 the filename and download options are being ignored. I'm using DomPdf v0.8.2 as engine.

Here is the config from boostrap.php:
```
Configure::write('CakePdf', [
    'engine' => [
    	'className' => 'CakePdf.DomPdf',
	    'options' => [
		    'isRemoteEnabled' => true
	    ]
    ],
    'margin' => [
	    'bottom' => 40,
	    'left' => 30,
	    'right' => 30,
	    'top' => 35
    ],
	'encoding'=>'UTF-8'
]);
```

And here is the setting from template:
```
$this->viewBuilder()->setOptions([
    'pdfConfig' => [
	    'orientation' => 'landscape',
	    'filename' => \Cake\Utility\Text::slug($entity->field) . '-' . __('Report') . '.pdf',
	    'download' => true
    ]
]);
```
Am I doing something wrong? Thanks for your help!

**Comment by T0mi** [comment]
Tried versions between 3.3.1 and 3.4.0. 

Version 3.3.2 is working fine, 3.3.3 will break on getType() function, and the problem is present in versions 3.3.4 and 3.4.0.

**Comment by ADmad** [comment]
I am pretty sure the required headers are set here https://github.com/FriendsOfCake/CakePdf/blob/423be26389b329a9edee7a8c82c8c80b072a4360/src/View/PdfView.php#L125

Can you debug and find out why do they not end up in response that is send?

**Comment by ADmad** [comment]
Or perhaps the `pdfConfig` view option is never passed to `CakePdfView` since it's not set as a valid view option.

**Comment by ADmad** [comment]
Hmm.. Then need to check whether same response instance is set to controller after rendering is completed.

**Comment by ADmad** [comment]
I tested with latest app skeleton and using the CakePdf config you have provided with `pdf` extension parsing enabled in router the controller code you have shown results if pdf file being downloaded as expected with specified file name.

I also tested without pdf extension (in which case `viewBuilder()->setClassName('CakePdf.Pdf')`) is needed and that works fine too.

Hence closing as I am unable to reproduce the problem.
